.github/workflows/integration_tests.yaml

```diff
@@ -0,0 +1,80 @@
+name: Integration tests
+
+on:
+  pull_request:
+
+jobs:
+  integration-test-charm-no-runner:
+    name: Integration test charm with no runner
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup operator environment
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+          juju-channel: 3.1/stable
+          provider: lxd
+      - name: Enable br_netfilter
+        run: sudo modprobe br_netfilter
+      - name: Run Integration tests
+        run: |
+          tox -e integration -- \
+            --keep-models \
+            --path ${{secrets.E2E_TESTING_REPO}} \
+            --token ${{secrets.E2E_TESTING_TOKEN}} \
+            --token-alt ${{secrets.E2E_TESTING_TOKEN_ALT}} \
+            tests/integration/test_charm_no_runner.py
```

Comment:
Minor nitpick no need change: would this be better if it was `tox -e integration -- -k test_charm_no_runner`?

Reply:
Same for other tests

Reply:
I feel using the filename should be better as it would be an exact match. Using patterns has the risk of matching something unintended.